src/main/java/io/fabric8/jenkins/openshiftsync/ImageStreamWatcher.java

```diff
@@ -152,36 +165,29 @@ private synchronized void onInitialImageStream(ImageStreamList imageStreams) {
         }
     }
 
-    private Map<String,PodTemplate> podTemplates(ImageStream imageStream) {
-        Map<String,PodTemplate> results = new HashMap<String,PodTemplate>();
-        String isuuid = imageStream.getMetadata().getUid();
-        if (hasSlaveLabelOrAnnotation(imageStream.getMetadata().getLabels()))
-            results.put(isuuid,
-                    podTemplateFromData(imageStream.getMetadata().getName(),
+    private List<PodTemplate> podTemplates(ImageStream imageStream) {
+        List<PodTemplate> results = new ArrayList<PodTemplate>();
+        if (hasSlaveLabelOrAnnotation(imageStream.getMetadata().getLabels())) {
+            results.add(podTemplateFromData(imageStream.getMetadata().getName(),
                     imageStream.getStatus().getDockerImageRepository(),
                     imageStream.getMetadata().getLabels()));
+        }
         
         String namespace = imageStream.getMetadata().getNamespace();
         
+        // since we cannot create watches on ImageStream tags, we have to traverse
+        // the tags and look for the slave label
         for (TagReference tagRef : imageStream.getSpec().getTags()) {
-            String istkey = "";
-            if (tagRef.getFrom() != null)
-                istkey = tagRef.getFrom().getName();
-            if (istkey.contains("/")) {
-                String[] parts = istkey.split("/");
-                if (parts.length > 1)
-                    istkey = parts[1];
+            ImageStreamTag ist = null;
+            try {
+                ist = getAuthenticatedOpenShiftClient().imageStreamTags().inNamespace(namespace).withName(imageStream.getMetadata().getName() + ":" + tagRef.getName()).get();
+            } catch (Throwable t) {
+                logger.log(Level.FINE, "podTemplates", t);
             }
-            if (istkey.length() > 0) {
-                ImageStreamTag ist = getAuthenticatedOpenShiftClient().imageStreamTags().inNamespace(namespace).withName(istkey).get();
-                if (ist != null && hasSlaveLabelOrAnnotation(ist.getMetadata().getAnnotations())) {
-                    // for the key, we do isuuid + istuuid so we can distinguish ist entries 
-                    // during cleanup actions induced by IS watch events
-                    results.put(isuuid + ist.getMetadata().getUid(),
-                            this.podTemplateFromData(ist.getMetadata().getName(), 
-                            ist.getImage().getDockerImageReference(), 
-                            ist.getMetadata().getAnnotations()));
-                }
+            if (ist != null && hasSlaveLabelOrAnnotation(ist.getMetadata().getAnnotations())) {
+                results.add(this.podTemplateFromData(ist.getMetadata().getName(), 
```

Comment:
is ist.getMetadata().getName()'s value "imagestreamname:tagname"?  or is it just "tagname"?